docs/guides/migrating-from-camunda-platform-7.md

```diff
@@ -530,16 +548,12 @@ Here's an example of a good Java Delegate:
 @Component
 public class CreateCustomerInCrmJavaDelegate implements JavaDelegate {
 
-    @Autowired
-    private ObjectMapper objectMapper;
-
     @Autowired
     private CrmFacade crmFacade;
 
     public void execute(DelegateExecution execution) throws Exception {
         // Data Input Mapping
-        String customerDataJson = (String) execution.getVariable("customerData");
-        CustomerData customerData = objectMapper.readValue(customerDataJson, CustomerData.class);
```

Comment:
Why did you remove the ObjectMapper - how would the data mapping work in C7? As we typically discourage object serialization and Spin - I think most realistic is to have a JSON String and transform it yourself - that's why I added it there

Reply:
I see, however, Java objects can be serialized in `application/json` directly (which is using Spin implicitely). 

At the same time, saving a complex object as String is an anti-pattern (I also removed a similar snippet below)

Reply:
@berndruecker should we discuss this?

Reply:
I would not fight this battle now due to bandwith reasons - but I still think using Spin is a bad idea and I would prefer storing JSON as String variable (or byte[] - wasn't there a difference in size? I already forgot :-)) and do serialization myself (what ends up in the database is the same - isn't it?). 

But as we are not advising people that this is a best practice thing, but give guidance on migrating existing things, it might not matter too much any way (nobody would base a decision on how to do their C7 program on this code example)

Reply:
@jonathanlukas So just go ahead if this is the consulting viewpoint on it 👍

Reply:
@berndruecker in general yes. 

To remove the reference to Spin from the example, we could retrieve the Java object directly and rely on the serialisation defaults of the engine (which should be `application/json`).

And yes, there is a problem with the text2 column in the process variable (which will allow 4000 characters) that some customers were already facing.

This is why I do not like the approach of using a String... without this limitation, I would go for this solution as well.